modules/jwplayerRtdProvider.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * This module adds the jwplayer provider to the Real Time Data module (rtdModule)
+ * The {@link module:modules/realTimeData} module is required
+ * The module will allow Ad Bidders to obtain JW Player's Video Ad Targeting information
+ * The module will fetch segments for the media ids present in the prebid config when the module loads. If any bid
+ * requests are made while the segments are being fetched, they will be blocked until all requests complete, or the
+ * timeout expires.
+ * @module modules/jwplayerRtdProvider
+ * @requires module:modules/realTimeData
+ */
+
+import { submodule } from '../src/hook.js';
+import { config } from '../src/config.js';
+import { ajaxBuilder } from '../src/ajax.js';
+import { logError } from '../src/utils.js';
+import find from 'core-js-pure/features/array/find.js';
+
+const SUBMODULE_NAME = 'jwplayer';
+let requestCount = 0;
+let requestTimeout = 150;
+const segCache = {};
+let resumeBidRequest;
+
+/** @type {RtdSubmodule} */
+export const jwplayerSubmodule = {
+  /**
+     * used to link submodule with realTimeData
+     * @type {string}
+     */
+  name: SUBMODULE_NAME,
+  /**
+     * get data and send back to realTimeData module
+     * @function
+     * @param {adUnit[]} adUnits
+     * @param {function} onDone
+     */
+  getData: getSegments,
+  init
+};
+
+config.getConfig('realTimeData', ({realTimeData}) => {
+  const providers = realTimeData.dataProviders;
+  const jwplayerProvider = providers && find(providers, pr => pr.name && pr.name.toLowerCase() === SUBMODULE_NAME);
+  const params = jwplayerProvider && jwplayerProvider.params;
+  if (!params) {
+    return;
+  }
+  const rtdModuleTimeout = params.auctionDelay || params.timeout;
+  requestTimeout = rtdModuleTimeout === undefined ? requestTimeout : Math.max(rtdModuleTimeout - 1, 0);
+  fetchTargetingInformation(params);
+});
+
+submodule('realTimeData', jwplayerSubmodule);
```

Comment:
@patmmccann this is the line where we register as an RTD submodule

Reply:
ty! how do you like this format?

Reply:
Do you mean how do I like having the jwplayer module be a submodule of an existing module ? I think it's great for reusability and consistency across all rtd providers. Thank you for suggesting this approach.